core/internal/src/mill/internal/Util.scala

```diff
@@ -4,4 +4,12 @@ private[mill] object Util {
   def leftPad(s: String, targetLength: Int, char: Char): String = {
     char.toString * (targetLength - s.length) + s
   }
+
+  /**
+   * Format a failed count as a string to be used in status messages.
+   * Returns ", N failed" if count > 0, otherwise an empty string.
+   */
+  def formatFailedCount(count: Int): String = {
+    if (count > 0) s", $count failed" else ""
+  }
 }
```

Comment:
Let's move this to the `Execution` companion object since that's where it's used